ngraph_bridge/pass/transpose_sinking.cc

```diff
@@ -481,12 +481,19 @@ bool TransposeSinking::run_on_function(shared_ptr<ngraph::Function> f) {
     n->revalidate_and_infer_types();
   }
 
-  // make sure shapes are always materialized before results
+  const ngraph::ResultVector& results = f->get_results();
   for (auto r : results) {
+    // make sure shapes are always materialized before results
     NGRAPH_CHECK(
         r->get_shape() == r->get_input_shape(0) &&
             r->get_element_type() == r->get_argument(0)->get_element_type(),
         " op::Result = ", *r, ", Arg = ", *r->get_argument(0));
+
+    // make sure that after TransposeSinking pass the output_shape for Result
+    // does not change from the expected output_shape before the pass
+    NGRAPH_CHECK(r->get_output_shape(0) == orig_result_out_shape[r->get_name()],
+                 " op::Result = ", *r, " expected output shape = ",
+                 orig_result_out_shape[r->get_name()]);
```

Comment:
I can combine the checks, left those separate for ease of understanding.

Reply:
I think it looks okay as is.